### PR TITLE
CPU NMS kernels replacement 

### PR DESCRIPTION
**Pull request by SKholkin**
CPU NMS kernels didn't work right so it was replaced by torch function.

### CODE REVIEW

File: examples/object_detection/layers/box_utils.py [hunk 1]
```diff
@@ -188,7 +189,10 @@ def forward(ctx, boxes, scores, threshold, top_k=200):
             return torch.tensor([], dtype=torch.int), torch.tensor(0)
         if scores.dim() == 1:
             scores = scores.unsqueeze(1)
-        keep = EXTENSIONS.nms(torch.cat((boxes, scores), dim=1), threshold, top_k)
+        if boxes.device.type == 'cpu':
```
Comment (ljaljushkin) [inline examples/object_detection/layers/box_utils.py:192]: I'd use `is_cuda` method for the Tensor. Please also add type hints for the function to make things clear

**Review by ljaljushkin — CHANGES_REQUESTED** [review]

File: examples/object_detection/layers/box_utils.py (hunk 1 above)
Comment (SKholkin) [reply examples/object_detection/layers/box_utils.py]: Done

**Comment by SKholkin** [comment]
Also kernels replacement doesn't affect perfomance
Old NMS kernels:
Detect for batch: 10/39 161.455s
Detect for batch: 11/39 160.931s
Detect for batch: 12/39 160.994s
Detect for batch: 13/39 160.790s
Torch NMS:
Detect for batch: 10/39 160.492s
Detect for batch: 11/39 159.978s
Detect for batch: 12/39 160.210s
Detect for batch: 13/39 159.942s

File: examples/object_detection/layers/extensions/nms/nms.cpp [hunk 2]
```diff
@@ -4,8 +4,5 @@
 #include "nms.h"
 
 at::Tensor nms(const at::Tensor& boxes, const float nms_overlap_thresh, const int64_t top_k) {
-  if (boxes.type().is_cuda()) {
-    return nms_gpu(boxes, nms_overlap_thresh, top_k);
-  }
-  return nms_cpu(boxes, nms_overlap_thresh, top_k);
+  return nms_gpu(boxes, nms_overlap_thresh, top_k);
```
Comment (vanyalzr) [inline examples/object_detection/layers/extensions/nms/nms.cpp:7]: Do we need this empty wrapper anymore? Why not just leave the `nms_gpu` function?

**Review by vshampor — APPROVED** [review]

**Comment by ljaljushkin** [comment]
@vshampor, let's agree in the future to not merge Sergei's pull requests until there's explicit approval from me or @vanyalzr and all comments are resolved. Deal? 🤝 

@SKholkin, please continue work on it in a separate PR on top of the current `develop` branch.

**Comment by AlexKoff88** [comment]
How about the export to ONNX? At some point, we gonna migrate to the export of DetectionOutput as a subgraph with NMS, top-k, and other nodes. Is it possible to export this NMS to ONNX?

**Pull request merged**